evm/src/main/java/org/hyperledger/besu/evm/processor/AbstractMessageProcessor.java

```diff
@@ -124,7 +124,10 @@ private void clearAccumulatedStateBesidesGasAndOutput(final MessageFrame frame)
    *
    * @param frame The message frame
    */
-  private void exceptionalHalt(final MessageFrame frame) {
+  private void exceptionalHalt(final MessageFrame frame, final OperationTracer operationTracer) {
+    if (operationTracer != null) {
+      operationTracer.traceContextExit(frame);
+    }
```

Comment:
Note that in line 134 the state is set to `COMPLETED_FAILED` and this will result in context edit called twice